### PR TITLE
Added new entry regionMatches() in Kotlin

### LINKED ISSUE #3732: [Term Entry] Kotlin Strings .regionMatches()
### Reason/inspiration (optional)

We would like a new term entry on the `regionMatches()` method for Kotlin: [Strings](https://www.codecademy.com/resources/docs/kotlin/strings). The entry should go in a new file under `docs/content/kotlin/concepts/strings/terms/regionMatches/regionMatches.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```kotlin `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Atharva1723**
Resolves #3732
My Username:
ruby9940067347

### Type of Change
Added new term entry regionMatches() in Kotlin 

### Checklist

- [✔ ] All writings are my own.
- [✔ ] My entry follows the Codecademy Docs style guide.
- [ ✔] My changes generate no new warnings.
- [ ✔] I have performed a self-review of my own writing and code.
- [ ✔] I have checked my entry and corrected any misspellings.
- [✔ ] I have made corresponding changes to the documentation if needed.
- [✔ ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ✔] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [✔ ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/kotlin/concepts/regionMatches/regionMatches.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+---
+Title: 'regionMatches()'
```
Comment (avdhoottt) [inline content/kotlin/concepts/regionMatches/regionMatches.md:2]:
```suggestion
Title: '.regionMatches()'
```

File: content/kotlin/concepts/regionMatches/regionMatches.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+Returns true if the specified region in this char sequence is equal to the specified region in another char sequence.
+
+Parameters
+
+- thisOffset - the start offset in this char sequence of the substring to compare.
+- other - the string against a substring of which the comparison is performed.
+- otherOffset - the start offset in the other char sequence of the substring to compare.
+- length - the length of the substring to compare.
```
Comment (avdhoottt) [inline content/kotlin/concepts/regionMatches/regionMatches.md:40]: These lines contain Plagiarism, please remove it.

File: content/kotlin/concepts/regionMatches/regionMatches.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  - 'Methods'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+**regionMatches()** for char sequence checks if a specified region within a Kotlin char sequence matches the content of another char sequence.
+
+**regionMatches()** for string checks if a specified region within a Kotlin String matches the content of another string.
```
Comment (avdhoottt) [inline content/kotlin/concepts/regionMatches/regionMatches.md:18]:
```suggestion
**.regionMatches()** for char sequence checks if a specified region within a Kotlin char sequence matches the content of another char sequence. It also checks if a specified region within a Kotlin String matches the content of another string.
```

File: content/kotlin/concepts/regionMatches/regionMatches.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+Returns true if the specified region in this string is equal to the specified region in another string.
+
+Parameters
+
+- thisOffset - the start offset in this string of the substring to compare.
+- other - the string against a substring of which the comparison is performed.
+- otherOffset - the start offset in the other string of the substring to compare.
+- length - the length of the substring to compare.
```
Comment (avdhoottt) [inline content/kotlin/concepts/regionMatches/regionMatches.md:60]: Use H2 headings to divide the content for CharSequence and String.

Comment (avdhoottt) [inline content/kotlin/concepts/regionMatches/regionMatches.md:60]: Contains Plagiarism.

File: content/kotlin/concepts/regionMatches/regionMatches.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+## Example
+
+The following example uses the `regionMatches()`for char sequence to check if a specified region within a Kotlin char sequence matches the content of another char sequence. The result is then printed:
```
Comment (avdhoottt) [inline content/kotlin/concepts/regionMatches/regionMatches.md:64]:
```suggestion
The following example uses the `.regionMatches()` for the char sequence. The result is then printed:
```
The line was repeated many times.

File: content/kotlin/concepts/regionMatches/regionMatches.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        length = 5,
+        ignoreCase = true
+    )
+
+    if (result) {
+        println("The specified region in charSeq1 matches the content of charSeq2.")
+    } else {
+        println("The specified region in charSeq1 does not match the content of charSeq2.")
+    }
+}
```
Comment (avdhoottt) [inline content/kotlin/concepts/regionMatches/regionMatches.md:84]:
```suggestion
fun main() {
    val charSeq1: CharSequence = "Hello, CodeAcademy Members"
    val charSeq2: CharSequence = "code"

    val result = charSeq1.regionMatches(
        thisOffset = 7,
        other = charSeq2,
        otherOffset = 0,
        length = 4,
        ignoreCase = true
    )

    if (result) {
        println("The specified region in charSeq1 matches the content of charSeq2.")
    } else {
        println("The specified region in charSeq1 does not match the content of charSeq2.")
    }
}
```
The string "code" has a length of 4.

File: content/kotlin/concepts/regionMatches/regionMatches.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+This prints the following output:
+
+```shell
+The specified region in charSeq1 matches the content of charSeq2.
+```
+
+The following example uses the `regionMatches()`for string to check if a specified region within a Kotlin string matches the content of another string. The result is then printed:
```
Comment (avdhoottt) [inline content/kotlin/concepts/regionMatches/regionMatches.md:93]:
```suggestion
The following example uses the `.regionMatches()` for string. The result is then printed:
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@Atharva1723  I have mentioned some changed, please correct them.

**Comment by Atharva1723** [comment]
@avdhoottt I have made necessary changes

File: content/kotlin/concepts/regionMatches/regionMatches.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+```pseudo
+// regionMatches() method for CharSequence
+fun CharSequence.regionMatches(
+    thisOffset: Int,
+    other: CharSequence,
+    otherOffset: Int,
+    length: Int,
+    ignoreCase: Boolean = false
+): Boolean
```
Comment (avdhoottt) [inline content/kotlin/concepts/regionMatches/regionMatches.md:28]: Explain the parameters of .regionMatches() without any Plagiarism. Add it below this syntax.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@Atharva1723 I have suggested some content addition, please add it to the content.

File: content/kotlin/concepts/regionMatches/regionMatches.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+
+Parameters:
+
+- thisOffset - the starting position in this string for the substring to be compared.
+
+- other - the string to which a substring is compared.
+
+- otherOffset - the starting position in the other string for the substring to be compared.
+
+- length - the length of the substring to be compared.
```
Comment (avdhoottt) [inline content/kotlin/concepts/regionMatches/regionMatches.md:66]:
```suggestion
Parameters:

- thisOffset - the starting position in this string for the substring to be compared.
- other - the string to which a substring is compared.
- otherOffset - the starting position in the other string for the substring to be compared.
- length - the length of the substring to be compared.
```

File: content/kotlin/concepts/regionMatches/regionMatches.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+Parameters:
+
+- thisOffset: This is the starting position within the current character sequence for the substring being compared.
+
+- other: This represents the string against which a substring comparison is carried out.
+
+- otherOffset: This indicates the starting position within the other character sequence for the substring being compared.
+
+- length: This specifies the length of the substring under comparison.
```
Comment (avdhoottt) [inline content/kotlin/concepts/regionMatches/regionMatches.md:41]:
```suggestion
- thisOffset: This is the starting position within the current character sequence for the substring being compared.
- other: This represents the string against which a substring comparison is carried out.
- otherOffset: This indicates the starting position within the other character sequence for the substring being compared.
- length: This specifies the length of the substring under comparison.
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@Atharva1723 Make these changed and rest of the content is fine but the path of this entry is wrong. Check the [issue](https://github.com/Codecademy/docs/issues/3732) and the this entry to this path - docs/content/kotlin/concepts/strings/terms/regionMatches/regionMatches.md

**Review by Atharva1723 — COMMENTED** [review]
All Changes Done

File: content/kotlin/concepts/strings/terms/regionMatches/regionMatches.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+```
+
+Returns true if the specified region in this char sequence is equal to the specified region in another char sequence.
+
+Parameters:
+
+- thisOffset: This is the starting position within the current character sequence for the substring being compared.
+- other: This represents the string against which a substring comparison is carried out.
+- otherOffset: This indicates the starting position within the other character sequence for the substring being compared.
+- length: This specifies the length of the substring under comparison.
```
Comment (SSwiniarski) [inline content/kotlin/concepts/strings/terms/regionMatches/regionMatches.md:38]:
```suggestion
- `thisOffset`: This is the starting position within the current character sequence for the substring being compared.
- `other`: This represents the string against which a substring comparison is carried out.
- `otherOffset`: This indicates the starting position within the other character sequence for the substring being compared.
- `length`: This specifies the length of the substring under comparison.
```

File: content/kotlin/concepts/strings/terms/regionMatches/regionMatches.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+```
+
+Returns true if the specified region in this string is equal to the specified region in another string.
+
+Parameters:
+
+- thisOffset - the starting position in this string for the substring to be compared.
+- other - the string to which a substring is compared.
+- otherOffset - the starting position in the other string for the substring to be compared.
+- length - the length of the substring to be compared.
```
Comment (SSwiniarski) [inline content/kotlin/concepts/strings/terms/regionMatches/regionMatches.md:60]:
```suggestion
- `thisOffset`: The starting position in this string for the substring to be compared.
- `other`: The string to which a substring is compared.
- `otherOffset`: The starting position in the other string for the substring to be compared.
- `length`: The length of the substring to be compared.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Atharva1723 I did a second review and had a few comments.

**Review by SSwiniarski — APPROVED** [review]
@Atharva1723 Looks good. @avdhoottt can you approve this so we can merge.

**Review by avdhoottt — APPROVED** [review]
@SSwiniarski Yes, we can merge this PR.

**Pull request merged**